pkg/controllers/vdb/replicatedupgrade_reconciler.go

```diff
@@ -352,17 +353,34 @@ func (r *ReplicatedUpgradeReconciler) upgradeSandbox(ctx context.Context) (ctrl.
 	}
 	if updated {
 		r.Log.Info("update image in sandbox", "image", r.VDB.Spec.Image)
+
+		// Get the sandbox podfacts only to invalidate the cache
+		sbPFacts, err := r.getSandboxPodFacts(ctx, false)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		sbPFacts.Invalidate()
 	}
 	return ctrl.Result{}, nil
 }
 
 // waitForSandboxUpgrade will wait for the sandbox upgrade to finish. It will
 // continually check if the pods in the sandbox are up.
 func (r *ReplicatedUpgradeReconciler) waitForSandboxUpgrade(ctx context.Context) (ctrl.Result, error) {
-	// The way I think we should do this is to get podfacts for the sandbox.
-	// Then wait for the nodes to be up. Each time we find the nodes aren't up
-	// we should requeue using the upgrade requeue time (see GetUpgradeRequeueTimeDuration).
-	return ctrl.Result{}, errors.New("wait for sandbox upgrade is not yet implemented")
+	sbPFacts, err := r.getSandboxPodFacts(ctx, true)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	r.Log.Info("collected sandbox facts", "numPods", len(sbPFacts.Detail))
+	for _, pf := range sbPFacts.Detail {
+		r.Log.Info("sandbox pod fact", "pod", pf.name.Name, "image", pf.image, "up", pf.upNode)
+		if pf.image != r.VDB.Spec.Image || !pf.upNode {
+			r.Log.Info("Still waiting for sandbox to be upgraded")
```

Comment:
How do you wake up sandbox controller to drive the upgrade?

Reply:
Yeah, I probably need to include the SandboxUpgradeReconciler in here.

Reply:
Are you going to address this?

Reply:
Yes, I applied this comment. But it's done in a separate function. See the call to `MakeSandboxUpgradeReconciler` in `upgradeSandbox`. I didn't think waking up the sandbox controller should be done in the wait function.